src/org/mozilla/javascript/ScriptableObject.java

```diff
@@ -249,7 +249,7 @@ ScriptableObject getPropertyDescriptor(Context cx, Scriptable scope) {
             String fName = name == null ? "f" : name.toString();
             if (getter != null) {
                 if ( getter instanceof MemberBox ) {
-                    desc.defineProperty("get", new FunctionObject(fName, ((MemberBox)getter).member(), scope), EMPTY);
+                    desc.defineProperty("get", ((MemberBox) getter).asGetterFunction(fName, scope, ScriptableObject.getFunctionPrototype(scope)), EMPTY);
```

Comment:
Same here I wish that we didn't have to do "instanceof MemberBox" but I do realize that this code is very very old.

Reply:
See above - and instanceof is really fast in java since years now